.gitignore

```diff
@@ -18,3 +18,4 @@ spack*
 compile_commands.json
 settings.json
 .vscode*
+.cache*
```

Comment:
Feels like we almost should be adding even more stuff in there... I would like
```
.ccls
.ccls-cache/
build/
```
to be there too